### PR TITLE
App installation finders

### PR DESCRIPTION
**Pull request by joshpuetz**
Adds support for the following installation finders from the Github Apps preview API to Octokit:

- https://developer.github.com/v3/apps/#find-organization-installation
- https://developer.github.com/v3/apps/#find-repository-installation
- https://developer.github.com/v3/apps/#find-user-installation

### CODE REVIEW

**Comment by joshpuetz** [comment]
I'm not sure what's going on with the Travis build for Ruby 2: looks like it's impacting another recent PR  (https://github.com/octokit/octokit.rb/pull/1050, https://travis-ci.org/octokit/octokit.rb/jobs/416504205).

**Comment by joshpuetz** [comment]
Ah, looks like Public Suffix (https://github.com/weppos/publicsuffix-ruby) dropped support for Rubies < 2.1 with version 3.0, referenced in this PR https://github.com/octokit/octokit.rb/pull/948#partial-pull-merging

**Comment by joshpuetz** [comment]
Pinging @tarebyte and @kytrinyx for review, thanks!

File: lib/octokit/client/apps.rb [hunk 1]
```diff
@@ -77,6 +77,45 @@ def create_integration_installation_access_token(installation, options = {})
...
+      #
+      # @param org [String] Organization GitHub login
+      # @param options [Hash] A customizable set of options
+      #
+      # @see https://developer.github.com/v3/apps/#find-organization-installation
+      #
+      # @return [Sawyer::Resource] Installation information
+      def find_organization_installation(organization, options = {})
+        opts = ensure_api_media_type(:integrations, options)
+        get "orgs/#{organization}/installation", opts
```
Comment (tarebyte) [inline lib/octokit/client/apps.rb]: Can we use the `Organization.path` helper here?

```ruby
get "#{Organization.path(organization)}/installation", opts
```

File: lib/octokit/client/apps.rb [hunk 2]
```diff
@@ -77,6 +77,45 @@ def create_integration_installation_access_token(installation, options = {})
...
+      #
+      # @param repo [String] A GitHub repository
+      # @param options [Hash] A customizable set of options
+      #
+      # @see https://developer.github.com/v3/apps/#find-repository-installation
+      #
+      # @return [Sawyer::Resource] Installation information
+      def find_repository_installation(repo, options = {})
+        opts = ensure_api_media_type(:integrations, options)
+        get "repos/#{repo}/installation", opts
```
Comment (tarebyte) [inline lib/octokit/client/apps.rb]: Can we use the `Repository.path` helper here?

```ruby
get "#{Repository.path(repo)}/installation", opts
```

File: lib/octokit/client/apps.rb [hunk 3]
```diff
@@ -77,6 +77,45 @@ def create_integration_installation_access_token(installation, options = {})
...
+      #
+      # @param user [String] GitHub user login
+      # @param options [Hash] A customizable set of options
+      #
+      # @see https://developer.github.com/v3/apps/#find-user-installation
+      #
+      # @return [Sawyer::Resource] Installation information
+      def find_user_installation(user, options = {})
+        opts = ensure_api_media_type(:integrations, options)
+        get "users/#{user}/installation", opts
```
Comment (tarebyte) [inline lib/octokit/client/apps.rb]: Can we use the `User.path` helper here?

```ruby
get "#{User.path(user)}/installation", opts
```

**Comment by joshpuetz** [comment]
Sensible suggestions, updated. Thank @tarebyte!

**Comment by joshpuetz** [comment]
I think this is ready to merge, thanks @tarebyte!

**Pull request merged**